traced/traced.go

```diff
@@ -111,6 +128,7 @@ func (tc *TraceCollector) handleStream(s network.Stream) {
 		case nil:
 			tc.mx.Lock()
 			tc.buf = append(tc.buf, msg.Batch...)
+			tc.bufremote = s.Conn().RemoteMultiaddr()
```

Comment:
this is not safe, we can have batches from multiple peers in the buffer.

Reply:
Oh, you're right. Let me fix that.